drivers/soundwire/intel_init.c

```diff
@@ -301,17 +371,35 @@ static struct sdw_intel_ctx
 
 	return ctx;
 
-err:
-	ctx->count = i;
-	sdw_intel_cleanup(ctx);
+del:
+	for (j = 0; j < i; j++) {
```

Comment:
That "j < i" condition looks a somewhat scary, but it would indeed seem to be correct.

Reply:
I changed the logic to use separate loop counters in the two steps, that way the logic is more regular and easier to review.